### PR TITLE
Past Index Maps

### PR DESCRIPTION
**Pull request by nedSaf**
#199
#202

### CODE REVIEW

**Comment by nedSaf** [comment]
Homepage Only the latest season:
selection_015

Past index page:
with select to all the seasons + a Save as a PDF functionality 
selection_014

**Comment by niryariv** [comment]
nice! please add these: 
- [x] +1 zoom level to the default view (ie, if the current default is 12 make it 13)
- [x] 15px margin between the map and "Save to PDF" button

**Comment by nedSaf** [comment]
I Changed the button and gave it margin:
selection_016

When i give +1 zoom to the map it looks like this:
selection_017

Should i keep it as it is or with the new zoom ?

**Comment by nedSaf** [comment]
Loader appears when clicking on "Save as PDF":
selection_018

When finished a Download link appears:
selection_019

And this is the final PDF:
selection_020

@niryariv Comments on the PR ?

**Comment by nedSaf** [comment]
Added ILRI logo to partners page #170 
selection_021

**Comment by niryariv** [comment]
zoom level ok. no need for a pdf viewer - clicking the button should just open the PDF file in whatever the user+browser default is (in a new tab, download etc)

**Comment by nedSaf** [comment]
This is what the download button does, The browser blocks the JS from opening a new to tab for download/view, so We figured to add this button instead of opening the PDF in the same window.

The New zoom level is ok ?

**Comment by niryariv** [comment]
ok

File: ibli/modules/custom/ibli_data/ibli_data.module [hunk 1]
```diff
@@ -5,6 +5,94 @@
...
+    'type' => MENU_CALLBACK,
+  );
+
+  return $items;
+}
+
+/**
+ * Saves the image of the map as a PDF file.
+ *
+ * @print string.
```
Comment (amitaibu) [inline ibli/modules/custom/ibli_data/ibli_data.module]: `@return`

File: ibli/modules/custom/ibli_data/ibli_data.module [hunk 2]
```diff
@@ -5,6 +5,94 @@
...
+ * PDF file path.
+ */
+function ibli_data_save_pdf() {
+  if (isset($_POST['map']) && isset($_POST['period'])) {
```
Comment (amitaibu) [inline ibli/modules/custom/ibli_data/ibli_data.module]: create access callback.

``` php
function ibli_data_save_pdf_access() {
  return user_access('access content') && !empty($_POST['map']) && !empty($_POST['period']);
```

File: ibli/modules/custom/ibli_data/ibli_data.module [hunk 3]
```diff
@@ -5,6 +5,94 @@
...
+    // Determine page layout according to the map dimensions.
+    $pagelayout = array($map_width, $map_height);
+
+    $pdf = new TCPDF('L', 'px', $pagelayout, true, 'UTF-8', false);
+
+    // Set margins.
+    $pdf->SetMargins(0, 0, 0);
+
+    // Set auto page breaks.
+    $pdf->SetAutoPageBreak(false, 0);
```
Comment (amitaibu) [inline ibli/modules/custom/ibli_data/ibli_data.module]: => FALSE.

**Comment by nedSaf** [comment]
Merged.

**Pull request closed without merging**